tap_snowflake/sync_strategies/common.py

```diff
@@ -111,6 +122,16 @@ def generate_select_sql(catalog_entry, columns):
             escaped_columns.append(f'TO_VARCHAR({escaped_col}) as {escaped_col}')
         elif property_format == 'geography':
             escaped_columns.append(f'ST_ASTEXT({escaped_col}) as {escaped_col}')
+        # Castings below were added for WP-21311 to make sure that Snowflake import using s3 unload sync vs regular sync 
+        # uses the same select query.
+        elif property_format == 'time':
+            escaped_columns.append(f"TO_VARCHAR({escaped_col}, 'HH24:MI:SS.FF6') as {escaped_col}")
+        # Symon simply drops timezone info instead of casting to UTC. Snowflake's TO_TIMESTAMP_NTZ also has the same behavior.
+        elif property_format == 'date-time':
+            escaped_columns.append(f'TO_TIMESTAMP_NTZ({escaped_col}) as {escaped_col}')
+        # Symon treats all number type as double.
+        elif data_type == 'number':
+            escaped_columns.append(f'TO_DOUBLE({escaped_col}) as {escaped_col}')
```

Comment:
For large numbers or Decimal type with specific precision/scale, precision/scale could be lost as we cast to double. This is the original behaviour as we cast all numbers to double(float) in target.

e.g.)
18,446,744,073,709,551,615 --> 18,446,744,073,709,552,000
5.18446744073709551615 --> 5.184467440737095